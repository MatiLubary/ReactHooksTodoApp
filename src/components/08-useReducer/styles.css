* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

body {
    padding: 50px;
}

p {
    margin: 0;
    cursor: pointer;
}

.list-group-item {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.mRa {
    margin-right: auto
}

.complete {
    text-decoration: line-through;
}

.w100 {
    width: 100%;
}