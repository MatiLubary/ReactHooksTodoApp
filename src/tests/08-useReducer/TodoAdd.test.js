import { shallow, mount } from 'enzyme'
import React from 'react'
import { TodoAdd } from '../../components/08-useReducer/TodoAdd'

describe('tests in <TodoAdd />', () => {


    const handleAddTodo = jest.fn()

    const wrapper = shallow(
        <TodoAdd
            handleAddTodo = {handleAddTodo}
        />
    )
    
    test('should display properly', () => {

        expect(wrapper).toMatchSnapshot();

        
    })



    test('should not call handleAddTodo', () => {

        const formSubmit = wrapper.find('form').prop('onSubmit')

        formSubmit({ preventDefault(){} });

        expect( handleAddTodo ).toHaveBeenCalledTimes(0)
        
    })


    test('should call function handleAddTodo', () => {

        wrapper.find('input').at(0).simulate('change', {target: {name: 'description', value: 'Hola'}});
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault: ()=>{}});
        expect(handleAddTodo).toHaveBeenCalled();
        
        
    })
    
    



})
