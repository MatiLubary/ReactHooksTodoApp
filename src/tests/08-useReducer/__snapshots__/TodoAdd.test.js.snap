// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <TodoAdd /> should display properly 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="learn..."
      text="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2 w100"
      type="submit"
    >
      ADD
    </button>
  </form>
</Fragment>
`;
